core/node/api_server/src/web3/backend_jsonrpsee/middleware.rs

```diff
@@ -144,7 +145,11 @@ where
             ObservedRpcParams::Unknown
         };
         let call = self.method_tracer.new_call(method_name, observed_params);
-        WithMethodCall::new(self.inner.call(request), call)
+        let mut future = WithMethodCall::new(self.inner.call(request), call);
+        if TRACE_PARAMS {
+            future.alloc = Some(AllocationAccumulator::new(method_name));
+        }
```

Comment:
This opt-in tracking is to prevent potential performance regression (I haven't noticed one locally, but that doesn't say much).